internal/stats/request.go

```diff
@@ -0,0 +1,47 @@
+package stats
+
+import (
+	"github.com/edulinq/autograder/internal/config"
+	"github.com/edulinq/autograder/internal/log"
+	"github.com/edulinq/autograder/internal/timestamp"
+)
+
+type RequestMetric struct {
+	BaseMetric
+
+	Endpoint     string `json:"endpoint"`
+	Locator      string `json:"locator,omitempty"`
+	CourseID     string `json:"course,omitempty"`
+	AssignmentID string `json:"assignment,omitempty"`
+	UserEmail    string `json:"user,omitempty"`
+	IPAddress    string `json:ip-address`
+	Value        uint64 `json:"duration"`
```

Comment:
Why call this "Value"?
For things like course metrics, we don't know what will be stored in the field.
But here, we know what it will be.